### DISCUSSION TITLE
Using GUID to Pull an Elements Item No.

### DISCUSSION THREAD

**Discussion by ninarular** [cwapi3d/cwapi3dpython#140] [General]
I'm currently producing a plugin to use a CSV file to replace a set of element attributes within cadwork. The csv itself only has the Item number to match to the Cadwork file, however I've found that there isn't a way to use the GUID to find the item no, unlike the assembly number. How may I find the Item No. based on the GUID alone?

**Comment by Brunner246** [comment]
Hi @ninarular 
Item number = part number, or which number do u mean?

Identical elements can have the same part number. Therefore more than one element can be related to the part number.

```python
import element_controller as ec
import attribute_controller as ac
import cadwork

your_part_nr_list = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

all_identifiable_element_ids = ec.get_all_identifiable_element_ids()

part_nr_element_ids = dict()
for part_nr in your_part_nr_list:
    element_ids = filter(lambda element_id: ac.get_part_number(element_id) == part_nr, all_identifiable_element_ids)
    part_nr_element_ids[part_nr] = list(element_ids)
```
